### ISSUE TITLE
Change the CLI of opening haddocks in the browser

### ISSUE THREAD

**Issue by mgsloan** [commercialhaskell/stack#2025]
I think the behavior in https://github.com/commercialhaskell/stack/pull/2015 is a bit surprising.  What if instead:

having `--open` take a package identifier argument (version optional).  This argument would implicitly get treated as a target, and we'd open the docs.

Then, we can also have `stack haddock --open-index`, which could work even when the build step has nothing to build.

One reason I like this is that latter is that later on we can further overload it with opening the docs for a particular module, like `stack haddock --open Control.Lens`.  That said, I already plan to overload the target syntax for `stack ghci` with something similar.  Might consider putting it under `stack ghci --module Control.Lens` or something, though.

**Comment by silky** [comment]
opening via package identifiers sounds fantastic to me; it will be really amazing to get to the haddock docs for any installed package so quickly, especially, as mentioned in another thread, the doom that is the current hackage docs.

**Comment by mgsloan** [comment]
@silky Yup, it's pretty great!  You can already do `stack haddock --open pkg` to open the docs for that package, but you have to specify just that single component.
